internal/handlers/common/update.go

```diff
@@ -118,5 +122,77 @@ func UpdateDocument(doc, update *types.Document) (bool, error) {
 		}
 	}
 
-	return true, nil
+	return changed, nil
+}
+
+// ValidateUpdateOperators validates update statement.
+func ValidateUpdateOperators(update *types.Document) (err error) {
+	set, err := getUpdateOperatorDocument("$set", update)
+	if err != nil {
+		return err
+	}
+	_, err = getUpdateOperatorDocument("$setOnInsert", update)
+	if err != nil {
+		return err
+	}
+
+	inc, err := getUpdateOperatorDocument("$inc", update)
+	if err != nil {
+		return err
+	}
+
+	if set != nil && inc != nil {
+		for _, setKey := range set.Keys() {
+			if inc.Has(setKey) {
+				return NewWriteErrorMsg(
+					ErrConflictingUpdateOperators,
+					fmt.Sprintf("Updating the path '%[1]s' would create a conflict at '%[1]s'", setKey),
+				)
+			}
+		}
+	}
+
+	for _, updateOp := range update.Keys() {
+		switch updateOp {
+		case "$set",
+			"$setOnInsert",
+			"$inc":
+			// supported
+		default:
+			return NewWriteErrorMsg(
+				ErrFailedToParse,
+				fmt.Sprintf("Unknown modifier: %s. Expected a valid update modifier or pipeline-style "+
+					"update specified as an array", updateOp,
+				))
+		}
+	}
+	return nil
+}
+
+// getUpdateOperatorDocument gets document by key `op` and returns WriteError error if it is not a document.
```

Comment:
This function is quite unclear to me. I don't understand what's the purpose - not by the doc, nor by the function name, nor by the code of this function. 

I see it receives a document called `update` and it returns some other document. What is the difference between the document it receives and the document it returns?

Reply:
> This function is quite unclear to me. I don't understand what's the purpose - not by the doc, nor by the function name, nor by the code of this function.
> 
> I see it receives a document called `update` and it returns some other document. What is the difference between the document it receives and the document it returns?

You are quite right. There is no difference. I just needed to extract a function to make a check.

Reply:
> 

I thought about naming the function starting `check` but it does return the document.

Reply:
I still don't understand :) Which function do you need to extract? What is the result of this extract?

Reply:
> I still don't understand :) Which function do you need to extract? What is the result of this extract?

the result of is extract is the `$set` or `$inc` expression document

Reply:
My initial question was what is the difference between the doc this function receives and returns? Maybe a small example of a document could help me understand.

Reply:
> My initial question was what is the difference between the doc this function receives and returns? Maybe a small example of a document could help me understand.

It's a smaller document.

The update document from the driver 
```go
bson.D{
	{"$set", bson.D{{"foo", int32(12)}}},
	{"$inc", bson.D{{"foo", int32(1)}}},
	{"$setOnInsert", bson.D{{"value", math.NaN()}}},
},
```

The document that returned by a function for `$setOnInsert` operator:
```go
bson.D{{"value", math.NaN()}
```
i.e. for `$inc`:
```go
bson.D{{"foo", int32(1)}}
```

Reply:
Based on this example, I'd suggest calling this function `FilterDocumentByOperator` unless the current name is already in accordance with some existing naming convention.

Also, what is the use case for this function? As a MongoDB user, what query do I send so you need to filter my query this way?

Reply:
Although, `Filter` is not a good word here. It might be misinterpreted as something relevant to `FilterDocument` function.

Reply:
It's not about filtering here. You can think of it as it's a bigger update document that we need to break down into smaller parts to validate their correctness. For that purpose, we already have a set of functions called `GetOptionalParam`.